packages/flutter_tools/test/commands.shard/permeable/build_bundle_test.dart

```diff
@@ -107,6 +107,26 @@ void main() {
     FeatureFlags: () => TestFeatureFlags(),
   });
 
+  testUsingContext('bundle fails to build for Windows arm64 if feature is disabled', () async {
+    globals.fs.file(globals.fs.path.join('lib', 'main.dart')).createSync(recursive: true);
+    globals.fs.file('pubspec.yaml').createSync(recursive: true);
+    globals.fs.file('.packages').createSync(recursive: true);
+    final CommandRunner<void> runner = createTestCommandRunner(BuildBundleCommand(
+      logger: BufferLogger.test(),
+      bundleBuilder: FakeBundleBuilder(),
+    ));
+
+    expect(() => runner.run(<String>[
+      'bundle',
+      '--no-pub',
+      '--target-platform=windows-arm64',
+    ]), throwsToolExit());
+  }, overrides: <Type, Generator>{
+    FileSystem: fsFactory,
+    ProcessManager: () => FakeProcessManager.any(),
+    FeatureFlags: () => TestFeatureFlags(),
```

Comment:
Could this test run with `isWindowsEnabled: true` and a new test flag like `isWindowsArm64Enabled: false`?